src/main/com/fulcrologic/fulcro/application.cljc

```diff
@@ -181,13 +182,23 @@
                                       (= "ui" ident-ns)
                                       (str/starts-with? ident-ns "com.fulcrologic.fulcro.")))))))))
 
+(defn- check-root-query-valid [query]
+  (when (and #?(:clj true :cljs goog.DEBUG)
+             query
+             (false? (s/valid? ::eql/query query)))
+    (throw (ex-info (str "The composed root query is not valid EQL. See `(comp/get-query "
```

Comment:
Do not throw. I'd much rather see an error in the console log that has no stack trace, since the S.T. is useless. Also, if the check is wrong and the query right for some reason (bug) it won't break things.